controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -500,15 +522,47 @@ func (r *StorageClassClaimReconciler) reconcileProviderPhases() (reconcile.Resul
 	if err := r.get(r.storageConsumer); err != nil {
 		return reconcile.Result{}, err
 	}
-
+	profile := ""
+	// check claim status already contains the name of the resource. if not, add it.
 	if r.storageClassClaim.Spec.Type == "blockpool" {
 		r.cephBlockPool = &rookCephv1.CephBlockPool{}
-		r.cephBlockPool.Name = fmt.Sprintf("cephblockpool-%s", r.storageConsumer.Name)
 		r.cephBlockPool.Namespace = r.OperatorNamespace
+		for _, res := range r.storageClassClaim.Status.CephResources {
+			if res.Kind == "CephBlockPool" {
+				r.cephBlockPool.Name = res.Name
+				profile = r.storageClassClaim.Status.StorageProfile
+				break
+			}
+		}
+		if r.cephBlockPool.Name == "" {
+			r.cephBlockPool.Name = fmt.Sprintf("cephfilesystemsubvolumegroup-storageconsumer-%s-%s", r.storageConsumer.Name, generateUUID())
```

Comment:
should be ephblockpool-%s not cephfilesystemsubvolumegroup-storageconsumer-%s-%s

Reply:
Thanks @oritwas yes it should be `cephblockpool-%s` fixed it.